docs/configuration.md

```diff
@@ -41,6 +41,7 @@ The server configuration is used to define how the application runs its server.
 | `tls.key` | The path to the tls key. | |
 | `auth.audience` | The audience for the IDP. | |
 | `auth.issuer` | The issuer for the IDP. | |
+| `auth.allowNoDPoP` | If true, we allow access tokens that do not have DPoP bindings. | `false` |
```

Comment:
The only concern I have here is we need a general platform setting that will drive the sdk as well. Unless we can get that information from the IDP openid-configuration which didn't sound reliable from @mkleene. 

@dmihalcik-virtru @biscoe916  Any thoughts on this?

Reply:
Ok I can get on board with that. But can we change the name of the config option. 

- skipDPoPVerification
- bypassDPoPCheck
- disableDPoPValidation
- dpopDisabled

Reply:
Those are all OK but they seem inaccurate to me. What we are doing is allowing access tokens that are not DPoP bound. All of those suggest (to me) that we are going to allow DPoP that is wrong through.

Reply:
OpenSSH seems to utilizes the word `strict` when it comes to this kind of behavior (i.e. `StrictModes`, `StrictHostKeyChecking`, `GSSAPIStrictAcceptorCheck`) both on client and in `/etc/sshd/sshd_config`.

What I would encourage is to focus firstly on a standard config syntax to communicate various behaviors. In this way, customer understanding today and the future is derived by manuals and consistency.

For this specific problem, it seems that Keycloak uses a positive language which resolves to `yes|no`. [DPOP_BOUND_ACCESS_TOKENS](https://github.com/keycloak/keycloak/blob/main/server-spi-private/src/main/java/org/keycloak/protocol/oidc/OIDCConfigAttributes.java#L49)

The language in their documentation seems to align with our implementation, no?

> If the switch OAuth 2.0 DPoP Bound Access Tokens Enabled is off, the **client can still send** DPoP proof in the token request. In that case, {project_name} will verify DPoP proof and will add the thumbprint to the token. But if the switch is off, DPoP binding is not enforced by the {project_name} server for this client. It is recommended to have this switch on if you want to make sure that particular client always uses DPoP binding.
>
> [Ref](https://github.com/keycloak/keycloak/blob/071032a108bd9e9fce9e66d00c36d56bd4b334df/docs/documentation/server_admin/topics/clients/oidc/con-advanced-settings.adoc#L100:~:text=OAuth%202.0%20Demonstrating%20Proof%2Dof%2DPossession%20at%20the%20Application%20Layer)

Reply:
I think that any of these are fine. My only reservation (at the code level) is that `false` is the zero value for booleans so if this ever gets created without reading the docs carefully then we won't be verifying DPoP.

Reply:
I'm fine with this for now. Some suggestions for a false-first name for this would be:

- `dpopOptional`
- `allowUnboundTokens